### DISCUSSION TITLE
Shopping list organization options

### DISCUSSION THREAD

**Discussion by djodars89** [mealie-recipes/mealie#1733] [Feature Request]
The shopping list is currently being generated by the recipes ingredients and is pretty static in how it's being displayed.

We would love an option to:
1. Add tag & categories support for ingredients (e.g. put all vegetables in one category, that way you can efficiently pick everything in that category when at the store because they're usually all next to each other versus if the list is ordered alphabetically).
2. Hide the quantity of each ingredients (e.g. Show 'All-purpose flour' instead of '3 tablespoons all-purpose flour').

**Comment by michael-genson** [comment]
For number 1, this is _partially_ available, you can label your foods with categories (e.g. a produce label, meat label, etc.). I actually have a custom integration with Todoist that uses the label as a section. I agree that we should add some additional grouping functionality in Mealie! I'll try my hand at this, though frontend isn't my strong suit.

For number 2, you can do this per-ingredient by setting quantity to zero (it won't actually say zero, it will just disappear). Are you saying you want to do this for the whole list, or just certain ingredients?

**Comment by Spalfnaganalf1** [comment]
Not sure whether to create a new feature request or add to this one - but i'd like to see the ability to filter the shopping list by the recipe type. The reason for this is that sometimes i have a recipe that includes a sauce, that i may already have frozen or prepared, so before i go shopping, i want to remove those items from the recipe. Would be good to be able to filter based on the recipe so that i know i'm not accidentally removing ingredients from another recipe.

**Reply by michael-genson** [reply]
While you can't exactly filter, each ingredient shows its source recipe now, letting you achieve the same thing
